src/dropdown/dropdown.spec.ts

```diff
@@ -15,8 +15,41 @@ function getDropdownEl(tc) {
   return tc.querySelector(`[ngbDropdown]`);
 }
 
+function getMenuEl(tc) {
+  return tc.querySelector(`[ngbDropdownMenu]`);
+}
+
+const jasmineMatchers = {
+  toBeShown: function(util, customEqualityTests) {
+    return {
+      compare: function(actual, content?, selector?) {
+        const dropdownEl = getDropdownEl(actual);
+        const menuEl = getMenuEl(actual);
+        const isOpen = dropdownEl.classList.contains('show') && menuEl.classList.contains('show');
+
+        return {
+          pass: isOpen,
+          message: `Expected ${actual.outerHTML} to have the "show class on both container and menu"`
+        };
+      },
+      negativeCompare: function(actual) {
+        const dropdownEl = getDropdownEl(actual);
+        const menuEl = getMenuEl(actual);
+        const isClosed = !dropdownEl.classList.contains('show') && !menuEl.classList.contains('show');
+
+        return {
+          pass: isClosed,
+          message: `Expected ${actual.outerHTML} not to have the "show class both container and menu"`
+        };
+      }
+    };
+  }
+};
+
 describe('ngb-dropdown', () => {
+
   beforeEach(() => {
+    jasmine.addMatchers(jasmineMatchers);
```

Comment:
Will open a separate issue / PR for this as I've noticed other things in this test that need attention.